### PR TITLE
Add support for uploading a Stainless config file in addition to the OpenAPI spec file

### PR DESCRIPTION
**Pull request by dgellow**
This pull request does three main changes:
1. The endpoint `POST /api/spec` has been updated to support a `multipart/form-data` body, making it possible to provide both the spec file and a Stainless config file. When the new input `config_path` is provided both the spec and config files will be uploaded, otherwise only the spec file will be uploaded.
2. While still supported, the use of S3 pre-signed URLs isn't required anymore. I removed that logic, the github action now relies on a single request `POST /api/spec` instead of the previous 3-steps dance.

### CODE REVIEW

File: action.yml [hunk 1]
```diff
@@ -13,5 +13,8 @@ inputs:
   input_path:
     description: "Path to the OpenAPI file."
     required: true
+  config_path:
+    description: "To to the Stainless config file."
```
Comment (RobertCraigie) [inline action.yml:17]:
```suggestion
    description: "Path to the Stainless config file."
```

File: index.ts [hunk 2]
```diff
@@ -9,97 +9,52 @@ export async function main() {
...
-
-  if (!response.ok) {
-    const errorMsg = `Failed to decorate spec: ${response.statusText} ${response.text}`;
-    error(errorMsg);
-    throw Error(errorMsg);
-  }
-
-  info('Decorated spec');
-  return response.text();
+  return response;
```
Comment (RobertCraigie) [inline index.ts:57]: question: why remove the `response.ok` and `response.text()` code?

File: index.ts [hunk 3]
```diff
@@ -9,97 +9,52 @@ export async function main() {
...
   if (!response.ok) {
-    const errorMsg = `Failed to create a signed upload: ${response.statusText} ${response.text}`;
+    const errorMsg = `Failed to upload spec or config file: ${response.statusText} ${response.text}`;
     error(errorMsg);
     throw Error(errorMsg);
   }
 
-  return response.json() as unknown as SignedUploadResponse;
+  if (outputPath) {
+    const decoratedSpec = response.text();
```
Comment (RobertCraigie) [inline index.ts:24]: nit: I think you have to `await` this

File: index.ts (hunk 2 above)
Comment (RobertCraigie) [reply index.ts:57]: Oh sorry you just moved it and the diffs hard to read.

File: index.ts [hunk 4]
```diff
@@ -9,97 +9,52 @@ export async function main() {
...
-  await uploadSpec(specPath, signedUpload);
-
-  info('Decorating spec...');
   const response = await fetch('https://api.stainlessapi.com/api/spec', {
     method: 'POST',
+    body: formData,
     headers: {
       Authorization: `Bearer ${token}`,
-      'Content-Type': 'application/json',
+      'Content-Type': 'multipart/form-data',
```
Comment (RobertCraigie) [inline index.ts:54]: doesn't have to happen in this PR but I think we should send a header indicating that the upload came from the github action so we can monitor it and provide better logging

Comment (dgellow) [reply index.ts:54]: Good idea 👍

Thanks for the review too :)

Comment (dgellow) [reply index.ts:54]: `X-GitHub-Action: true`, or set a specific user agent? What do you think?

File: index.ts (hunk 3 above)
Comment (dgellow) [reply index.ts:24]: Surprised the type checker didn't report this issue.

File: index.ts (hunk 4 above)
Comment (dgellow) [reply index.ts:55]: I went with `"X-Github-Action": "stainless-api/upload-openapi-spec-action"`, but maybe the user agent header would make sense here 🤷

File: index.ts (hunk 2 above)
Comment (dgellow) [reply index.ts:59]: Yes, it's simpler to read the resulting file, the diff looks a bit messy.

**Comment by dgellow** [comment]
Tested this branch via a GitHub workflow, it seems to work but we get a deprecation warning:
```
Run stainless-api/upload-openapi-spec-action@sam/sta-2798-update-upload-github-action-to-support-a-stainless-config-in
  with:
    stainless_api_key: ***
    input_path: specs/sink-openapi.yml
Uploading spec and config files...
(node:1383) [https://github.com/node-fetch/node-fetch/issues/1167] DeprecationWarning: form-data doesn't follow the spec and requires special treatment. Use alternative package
(Use `node --trace-deprecation ...` to show where the warning was created)
```

I will look into it.

**Pull request merged**